Logging/.CSProject/LogHandler/AbstractLogHandler.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+
+namespace Anvil.CSharp.Logging
+{
+    /// <summary>
+    /// A type capable of handling logs received by <see cref="Log"/>.
+    /// </summary>
+    public abstract class AbstractLogHandler
+    {
+        /// <summary> Log format part - the log message itself </summary>
+        public const string LOG_PART_MESSAGE = "{0}";
+        /// <summary> Log format part - the caller's type name </summary>
+        public const string LOG_PART_CALLER_TYPE = "{1}";
+        /// <summary> Log format part - the caller's method name </summary>
+        public const string LOG_PART_CALLER_METHOD = "{2}";
+        /// <summary> Log format part - the caller's source file path </summary>
+        public const string LOG_PART_CALLER_PATH = "{3}";
+        /// <summary> Log format part - the caller's source file name </summary>
+        public const string LOG_PART_CALLER_FILE = "{4}";
+        /// <summary> Log format part - the caller's source file line number </summary>
+        public const string LOG_PART_CALLER_LINE = "{5}";
+        /// <summary> Log format part - the timestamp at which the log was emitted. See <see cref="TimestampFormat"/> to customize format. </summary>
+        public const string LOG_PART_TIMESTAMP = "{6}";
+        /// <summary> Log format part - the log level. </summary>
+        public const string LOG_PART_LOG_LEVEL = "{7}";
```

Comment:
Interesting  I didn't know you could opt into these with string.Format. I assumed it was more "dumb" and expected params in order {0}, {1}, {etc} and that the number of actual values you passed in had to match.

Reply:
It does still work that way, the `string.Format(...)` in `HandleLog` is passed each of the strings that correspond to these constants every time, so their position doesn't change (ex. the original message is always passed first).

Reply:
Yeah sorry I meant I didn't think you could have "holes" in your formatting. 

If you pass a string with {0}, {3}, {5} I didn't think that would work.

Reply:
Oh gotcha, yeah they don't have to all be present, or even in order or only included once.